.github/workflows/check-duckdb-schema.yml

```
@@ -0,0 +1,57 @@
+# Verifies if our duckdb schema in R2 is up to date with the latest changes.
+# If it's not up to date, it will regenerate the file and upsert it.
+name: check-duckdb-schema
+env:
+  SQLMESH_DUCKDB_LOCAL_PATH: "/tmp/oso.duckdb"
+  GOOGLE_PROJECT_ID: "opensource-observer"
+
+# For now this only runs on a schedule once a day
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+  schedule:
+    # Schedule every day at 00:00 UTC
+    - cron: "0 0 * * *"
+  push:
+    branches: 
+      - "IcaroG/duckdb-schema"
+
+jobs:
+  check-duckdb-schema:
+    name: check-duckdb-schema
+    environment: deploy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: "Setup Python, Poetry and Dependencies"
+        uses: packetcoders/action-setup-cache-python-poetry@main
+        with:
+          python-version: 3.12
+          poetry-version: 1.8.2
+
+      - name: "Set up Cloud SDK"
+        uses: "google-github-actions/setup-gcloud@v2"
+        with:
+          version: ">= 363.0.0"
+
+      - name: Authenticate to google
+        uses: 'google-github-actions/auth@v2'
+        with:
+          credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_JSON }}'
+          create_credentials_file: true
+
+      - name: Download up to date schema
+        run: poetry run oso metrics local initialize --max-results-per-query 1 --max-days 1
+
+      - name: Upload file to Cloudflare R2
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.CLOUDFLARE_R2_S3_API_ACCESS_KEY }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.CLOUDFLARE_R2_S3_API_SECRET_KEY }}
+          AWS_ENDPOINT_URL: ${{ secrets.CLOUDFLARE_R2_S3_API_ENDPOINT }}
+          BUCKET_NAME: "oso-public"
+        run: bash .github/scripts/write-s3.sh "$SQLMESH_DUCKDB_LOCAL_PATH" "oso.duckdb"
```

I'd also pass in the endpoint here, just to be explicit.

```suggestion
        run: bash .github/scripts/write-s3.sh "$SQLMESH_DUCKDB_LOCAL_PATH" "oso.duckdb" "$CLOUDFLARE_R2_S3_API_ENDPOINT"
```

---

Done!